### PR TITLE
[Query] Make pool incentives query show % of incentives going to that gauge ID

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #1569

## What is the purpose of the change

Make pool incentives query show % of incentives going to that gauge ID
1. A pool has many gauges
2. Each gauge is a specific lockup duration in the pool
3. Therefore,  a Pool incentive % is the (gauge_id_weight / sum_of_all_pool_gauge_weight) * 100

Apologies this took a while, got a lil busy with other stuff 

## Brief Changelog
- amends the current query to show incentive percentage 

## Testing and Verifying
n/a

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (not applicable)

### CODE REVIEW

File: x/gamm/pool-models/balancer/tx.pb.go [hunk 1]
```diff
@@ -151,34 +151,34 @@ func init() {
 }
```
Comment (mattverse) [inline x/gamm/pool-models/balancer/tx.pb.go:152]: Note to reviewers: this change comes from uncommited changes from main

File: x/superfluid/types/query.pb.go [hunk 2]
```diff
@@ -1267,105 +1267,105 @@ func init() {
 func init() { proto.RegisterFile("osmosis/superfluid/query.proto", fileDescriptor_e3d9448e4ed3943f) }
 
 var fileDescriptor_e3d9448e4ed3943f = []byte{
```
Comment (mattverse) [inline x/superfluid/types/query.pb.go:1269]: Note to reviewers: this change comes from uncommited changes from main

File: x/pool-incentives/keeper/grpc_query.go [hunk 3]
```diff
@@ -32,18 +32,30 @@ func (q Querier) GaugeIds(ctx context.Context, req *types.QueryGaugeIdsRequest)
...
+		}
+
 		gaugeIdsWithDuration[i] = &types.QueryGaugeIdsResponse_GaugeIdWithDuration{
-			GaugeId:  gaugeId,
-			Duration: duration,
+			GaugeId:                  gaugeId,
+			Duration:                 duration,
+			GaugeIncentivePercentage: incentivePercentage.String(),
 		}
+
```
Comment (mattverse) [inline x/pool-incentives/keeper/grpc_query.go:58]:
```suggestion

```

**Review by mattverse — CHANGES_REQUESTED** [review]
Can we add test cases for this query before we merge this? Main logic looks good to me but we definitely need test cases for this to confirm! 

We should also revert `go.mod`!

File: x/pool-incentives/keeper/grpc_query.go [hunk 4]
```diff
@@ -32,18 +32,30 @@ func (q Querier) GaugeIds(ctx context.Context, req *types.QueryGaugeIdsRequest)
...
 	for i, duration := range lockableDurations {
 		gaugeId, err := q.Keeper.GetPoolGaugeId(sdkCtx, req.PoolId, duration)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 
+		for _, record := range distrInfo.Records {
+			if record.GaugeId == gaugeId {
+				// Pool incentive % = (gauge_id_weight / sum_of_all_pool_gauge_weight) * 100
+				incentivePercentage = record.Weight.ToDec().Quo(totalWeightDec).MulInt(sdk.NewInt(100))
```
Comment (alexanderbez) [inline x/pool-incentives/keeper/grpc_query.go:49]: I would put `sdk.NewInt(100)` into a var so we don't have to keep allocating memory for it.

**Review by alexanderbez — CHANGES_REQUESTED** [review]

File: x/pool-incentives/keeper/grpc_query_test.go [hunk 5]
```diff
@@ -285,3 +285,46 @@ func (suite *KeeperTestSuite) TestIncentivizedPools2() {
...
+		Weight:  sdk.NewInt(300),
+	})
+	suite.NoError(err)
+
+	res, err := queryClient.GaugeIds(context.Background(), &types.QueryGaugeIdsRequest{
+		PoolId: poolId,
+	})
+
+	suite.NoError(err)
+	suite.Equal("16.666666666666666700", res.GaugeIdsWithDuration[0].GaugeIncentivePercentage)
```
Comment (stackman27) [inline x/pool-incentives/keeper/grpc_query_test.go:327]: @mattverse @alexanderbez what's the best way of rounding these numbers or should is that something frontend handles?

**Comment by stackman27** [comment]
@alexanderbez @mattverse  Thank you for the review! 

Added test, lmk if theres anything else to address

File: x/pool-incentives/keeper/grpc_query_test.go (hunk 5 above)
Comment (mattverse) [reply x/pool-incentives/keeper/grpc_query_test.go:327]: Any chain logic should be completely independent from front end logic. That being said, there would be no need to round these numbers.

**Review by mattverse — APPROVED** [review]
Nice job on the test cases! 
Good to merge for now, we should make these tests tdd in the future

**Comment by stackman27** [comment]
> tdd

Yup planning on working on that based on priorities

File: x/pool-incentives/keeper/grpc_query_test.go (hunk 5 above)
Comment (stackman27) [reply x/pool-incentives/keeper/grpc_query_test.go:327]: Gotchu

Comment (stackman27) [reply x/pool-incentives/keeper/grpc_query_test.go:327]: Gotchu

**Review by p0mvn — APPROVED** [review]

**Review by ValarDragon — APPROVED** [review]
Awesome! Lets add a changelog entry and get this merged?

**Comment by stackman27** [comment]
> Awesome! Lets add a changelog entry and get this merged?

Should i add it in this PR or a separate one?

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**